tensorflow/tools/docker/Dockerfile.devel-rocm

```diff
@@ -0,0 +1,227 @@
+# This Dockerfile provides a starting point for a ROCm installation of 
+# MIOpen and tensorflow.
+
+# This Dockerfile uses a multi-stage build
+# The first stage is to build the HCC, HIP and other tools we need for the TF build
+# The second stage is to do the TF CI build itself
+# The separation of stages allows to reduce the size of the final docker image
+# by copying over only the packages built in the first stage over to the second one
+
+
+###################################################
+# Stage 1 : build the tools needed for the TF build
+###################################################
+
+FROM rocm/rocm-terminal:1.8.2 as tool_builder
+
+ENV DEBIAN_FRONTEND noninteractive
+ENV HOME /home/rocm-user
+
+# install packages needed for this image
+RUN sudo apt-get update && sudo apt-get install -y rpm git mercurial libxml2 libxml2-dev 
+
+# Workaround : build HCC from source using the commit that is known to have fix for issue812
+RUN cd $HOME && git clone --recursive https://github.com/RadeonOpenCompute/hcc.git
+RUN cd $HOME/hcc  && git checkout -b issue812-fix d530be7
```

Comment:
@parallelo could you remind me what issue 812 is?

Reply:
I'm not fully up-to-date on that particular topic -- best answered by @deven-amd.  This Dockerfile partially came from the one under `ci_build/Dockerfile.rocm` (which originally came from me, but it was modified extensively by Deven regarding MIOpen workarounds)

Reply:
@deven-amd Can you please give an update on these three specific workarounds from [here](https://github.com/ROCmSoftwarePlatform/tensorflow-upstream/blob/develop-upstream/tensorflow/tools/ci_build/Dockerfile.rocm#L11-L51)?  
HCC:  
```
# Workaround : build HCC from source using the commit that is known to have fix for issue812
RUN cd $HOME && git clone --recursive https://github.com/RadeonOpenCompute/hcc.git
RUN cd $HOME/hcc  && git checkout -b issue812-fix d530be7
```
HIP:  
```
 # Workaround: build HIP from source using fork that holds roc-1.8.x with the alternate fix for HIP PR 457
RUN cd $HOME && git clone -b roc-1.8.x-pr457-altfix https://github.com/deven-amd/HIP.git
```
MIOpen:  
```
# Workaround : build MIOpen from source using fork of the public repo with the temporary fix for issue #1061 ($HOME env not set)
RUN cd $HOME && git clone -b pr1061-fix https://github.com/deven-amd/MIOpen.git
```

Reply:
Issue 812 refers to the following :
https://github.com/RadeonOpenCompute/hcc/issues/812

HIP workaround is to use the "alternate fix" for HIP PR457

MIOpen workaround is to use "/tmp" as the home dir when $HOME is not set.

I believe all 3 workarounds will become redundant once we move to ROCm 1.9